14_final_schema.sql

```diff
@@ -73,6 +75,15 @@ USING(subject_id, hadm_id, stay_id, chart_hour)
 FULL JOIN
 `elevated-pod-307118.physionet.iv_antibiotics`
 USING(subject_id, hadm_id, stay_id, chart_hour)
+-- iv_abx_pharmacy is rolled up to the stay x chart_hour x medication grain 
+-- we need to roll it up further to the stay x chart_hour grain in order to join with the rest of the tables
+FULL JOIN
+(
+    select subject_id, hadm_id, stay_id, chart_hour, count(1) as abx_orders
+    FROM `elevated-pod-307118.physionet.iv_abx_pharmacy`
+    GROUP BY subject_id, hadm_id, stay_id, chart_hour
```

Comment:
This is where I'm still confused - are `subject_id`, `hadm_id`, and `stay_id` all unique? If so, then there's no reason for grouping by all three - only `stay_id` should be needed. To make it concrete, there are two cases:

1. The stay id is not unique between patients/admissions. For example, there could be two subjects - 1 and 2. Each subject then has their admission id start at 1 and stay id start at 1.
2. The stay id _is_ unique between patients/admissions. For example, subjects 1 and 2 have admissions 1234 and 1235 and stay ids 56789 and 56790, respectively. 

If it's case 2, then just group by stay id. If it's case 1, then that's definitely unexpected. I know this has been addressed previously, but I wanted to double-check.

Reply:
Not sure I'm right here, but while it's true that stay_id is all we need to join the records, I believe we still need to group by all of these fields since we want to keep the subject_id and hadm_id fields in the final table.

I think upcoming changes to the queries will include just joining on subject_id and chart_hour, and then having hadm_id and stay_id as optional fields - this is so we can capture records in the ED.

That's not included in this PR because this branch was meant for solely incorporating additional abx fields

Reply:
That's a good point about `group by`! I overlooked that. There is a way around this - the `count(1)` becomes a further subquery while joining with other fields. The `group by` with the other columns isn't ideal, but that's better than a complicated query that's hard to understand. (I haven't gone back to think through whether the upcoming changes will help or not, so I don't have good feedback on that right now.)

For the scope of the PR, I looked at the entire commit history - there are some older commits in here, so I just reviewed everything instead of the single commit. The `group by` is part of the PR, so it got reviewed. My recommendation would be to get the PR closed asap, and if there are leftover issues then add them as `TODO`s in the code. Tech debt is understandable sometimes, especially if the code works but isn't quite optimal.